keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -638,24 +640,107 @@ always dispatched immediately.  Following is how the other requests
 are dispatched at a given apiserver.
 
 The concurrency limit of an apiserver is divided among the non-exempt
-priority levels in proportion to their assured concurrency shares.
-This produces the assured concurrency value (ACV) for each non-exempt
-priority level:
+priority levels, and higher ones can do a limited amount of borrowing
+from lower ones.
+
+Non-exempt priority levels are ordered in a total order for the
+purpose of borrowing currently-unused concurrency.  The ordering is
+based (first) on increasing value of a spec field whose name is
+`priority` and whose value is constrained to lie in the range 0
+through 10000 inclusive and (second) on increasing name of the
+PriorityLevelConfiguration object.  Priority levels that appear later
```

Comment:
If you have to have a total order, instead of attempting to force humans to use unique numbers for everything (a nightmare if you ever want to incorporate configs from multiple places), please order first by priority, and then break ties alphabetically by name. If you don't want it to be predictable, you can hash one or both of those.

(To keep the actual comparisons numeric & easy, I'd expect to convert whatever the tie break is to a number and treat it as a fractional part of the priority number, but that's just implementation details)

Reply:
Since kube api-machinery has no multi-object transactions there is no direct way to prevent collisions.  We *could* do a validation check that catches violations that are well enough separated in time.  We *could* add a controller that complains somehow or deletes violating objects, but I think that would be more code to maintain than it is worth.  I have revised to more strongly say "don't do that" and not promise how ties are broken.